### PR TITLE
Add documentation for TypeScript usage

### PR DESCRIPTION
**Pull request by kibertoad**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,6 +85,35 @@ function worker (arg, cb) {
...
+function worker (arg: Task, cb: done) {
+  return Promise.resolve()
+          .then(() => {
+            console.log(arg.id)
+            cb(null)
+          })
+          .catch((err) => {
+            console.error(err)
+            cb(err)
+          })
```
Comment (mcollina) [inline README.md:113]: Can you please remove this `new Promise`?

Also, could you add an example for the promise api of fastq?

**Review by mcollina — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (kibertoad) [reply README.md]: Done.

File: README.md [hunk 2]
```diff
@@ -85,6 +84,53 @@ function worker (arg, cb) {
...
+type Task = {
+  id: number
+}
+
+const q: queueAsPromised<Task> = fastq.promise(asyncWorker, 1)
+
+q.push({ id: 42})
+
+async function asyncWorker (arg: Task): Promise<void> {
+  try {
```
Comment (kibertoad) [inline README.md:126]: @mcollina Do you prefer try-catch or Promise.resolve().then().catch() here? Is it generally safe not to catch a promise rejection here, will that end up as an unhandled rejection or fastq handles that itself?

Comment (mcollina) [reply README.md:126]: This is catched by fastq, there is no need to add a try-catch here: https://github.com/mcollina/fastq/blob/218eac7cb0a7c65cf4a81c3c370d269d29ac7a2b/queue.js#L216.

Comment (kibertoad) [reply README.md]: Updated!

File: README.md [hunk 3]
```diff
@@ -85,6 +84,50 @@ function worker (arg, cb) {
...
+import * as fastq from "fastq";
+import type { queueAsPromised } from "fastq";
+
+type Task = {
+  id: number
+}
+
+const q: queueAsPromised<Task> = fastq.promise(asyncWorker, 1)
+
+q.push({ id: 42})
```
Comment (mcollina) [inline README.md:123]:
```suggestion
q.push({ id: 42}).catch((err) => console.error(err))
```

Comment (mcollina) [reply README.md:123]: wdyt?

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**